ee/modules/021-cni-cilium/hooks/ee/helpes.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2024 Flant JSC
+Licensed under the Deckhouse Platform Enterprise Edition (EE) license. See https://github.com/deckhouse/deckhouse/blob/main/ee/LICENSE
+*/
+
+package ee
+
+import "github.com/flant/addon-operator/pkg/module_manager/go_hook"
+
+func Filter[T any](slice []T, fn func(T) bool) []T {
+	result := make([]T, 0, len(slice))
+
+	for _, v := range slice {
+		if fn(v) {
+			result = append(result, v)
+		}
+	}
+
+	return result
+}
+
+func MakeSliceFromSnapshots[T any](snapshots []go_hook.FilterResult) []T {
+	result := make([]T, 0, len(snapshots))
```

Comment:
This is an allocation. Each time calling this function you create a full copy of your input data.

Reply:
We don't use complex item and it is a more convenient way than type assertion in for...range

Reply:
Not a zero cost abstraction though. It looks like a complication without benefits (at least for now).

Reply:
It has better performance. If we omit capacity than there would be 3 automatic reslicng for input data with length equal 5